### ISSUE TITLE
Republishing a page not working

### ISSUE THREAD

**Issue by CrispinF** [cloudscribe/cloudscribe.SimpleContent#289]
This might be the result of some caching, not sure. 
Take any published page, and set it's Publication Date into the future. It disappears for public users as expected. Set the date back into the past - the page does not re-appear. e.g. I have this page and it's not appearing:
![image](https://user-images.githubusercontent.com/10574730/41206109-d521f552-6cf6-11e8-9356-8bcc34f977d4.png)
There are not view roles protecting the page, and it was visible immediately before the changes.

**Comment by CrispinF** [comment]
Update: set the checkbox "Is Published" to False > Save > set it to True again > Save > page now appears.

**Comment by joeaudette** [comment]
I'm not having luck trying to reproduce this problem. Can you confirm the data layer used and the cloudscribe versions under Administration > System Information?

I did find a different bug where the menu cache was not being cleared if the pubdate changed so a page with the pubdate changed to the future was not dissapearing from the menu, clicking the menu item was correctly resulting in 404 for anonymous requests.

I've fixed that issue in cloudscribe.SimpleContent.Web 3.0.1

**Comment by CrispinF** [comment]
@joeaudette  Reproduced on staging site https://highland.esdm.co.uk with versions:

Operating System
Microsoft Windows 6.1.7601 S X64
Runtime
.NET Core 4.6.26212.01
Environment Name
Production
Database Platform
Entity Framework with Microsoft SqlServer
cloudscribe.Core.Web
2.1.16
cloudscribe.Core.Identity
2.0.19
cloudscribe.Core.Storage.EFCore.MSSQL
2.0.12
cloudscribe.FileManager.Web
2.0.11
cloudscribe.Logging.Web
2.1.3
cloudscribe.SimpleContent
2.0.22
cloudscribe.SimpleContent.Storage.EFCore.MSSQL
2.0.11
cloudscribe.SimpleContent.Web.Mvc
2.0.22
cloudscribe.Web.Common
2.0.37

I've given you an account,  u/n & pwd coming by separate comms.

**Comment by joeaudette** [comment]
@CrispinF given that this does not appear to be a problem in the latest cloudscribe SimpleContent and you have figured out a workaround. I'm not sure what you are expecting me to do. It doesn't seem like an urgent issue that needs a fix for SimpleContent 2.x since updating to the latest version should solve it.

**Comment by joeaudette** [comment]
@CrispinF there was a bug found related to this just not exactly what you described. The problem I found was the page did not disappear from the menu for anonymous users if you updated it to a future pub date. The page could not be reached by anonymous users but the menu cache was not cleared. That has been fixed and updated 3.0.x nuget published. I am satisfied this is fixed but the issue can remain open until you have a chance to upgrade and verify the fix.
